src/groupeddataframe/splitapplycombine.jl

```diff
@@ -1090,10 +1027,32 @@ function _agg2idx_map_helper(idx, idx_agg)
 end
 
 function _combine(f::AbstractVector{<:Pair},
-                  gd::GroupedDataFrame, nms::AbstractVector{Symbol})
+                  gd::GroupedDataFrame, nms::AbstractVector{Symbol},
+                  copycols::Bool, keeprows::Bool) # TODO: remove these defaults
     # here f should be normalized and in a form of source_cols => fun
     @assert all(x -> first(x) isa Union{Int, AbstractVector{Int}, AsTable}, f)
     @assert all(x -> last(x) isa Base.Callable, f)
+
+    if keeprows
+        if !_check_cannonical(gd)
```

Comment:
Should be "canonical".

Reply:
I just removed this function and replaced it with `minimum(gd.groups) == 0` as it is easy enough to digest.